module/zfs/zpl_super.c

```diff
@@ -249,13 +249,46 @@ zpl_fill_super(struct super_block *sb, void *data, int silent)
 }
 
 #ifdef HAVE_MOUNT_NODEV
```

Comment:
Since we're no longer using `HAVE_MOUNT_NODEV` we should update this configure check to instead check for the `file_system_type->mount` prototype.   How about replacing `config/kernel-mount-nodev.m4` with `config/kernel-fst-mount.m4` and updating the macros accordingly.

```sh
dnl #
dnl # 2.6.38 API change
dnl # The .get_sb callback has been replaced by a .mount callback
dnl # in the file_system_type structure.
dnl #
AC_DEFUN([ZFS_AC_KERNEL_FST_MOUNT], [
        AC_MSG_CHECKING([whether fst->mount() exists])
        ZFS_LINUX_TRY_COMPILE([
                #include <linux/fs.h>

                static struct dentry *
                mount(struct file_system_type *fs_type, int flags,
                    const char *osname, void *data) {
                        struct dentry *d = NULL;
                        return (d);
                }

                static struct file_system_type fst __attribute__ ((unused)) = {
                        .mount = mount,
                };
        ],[
        ],[
                AC_MSG_RESULT(yes)
                AC_DEFINE(HAVE_FST_MOUNT, 1, [fst->mount() exists])
        ],[
                AC_MSG_RESULT(no)
        ])
])
```